core/src/com/unciv/ui/screens/worldscreen/WorldMapHolder.kt

```diff
@@ -230,6 +230,8 @@ class WorldMapHolder(
     }
 
     private fun onTileRightClicked(unit: MapUnit, tile: Tile) {
+        if (!UncivGame.Current.settings.singleTapMove) return
```

Comment:
> ...Yes very easy: Worldmapholder line 145-157. Move the entire line to just before the comment and continue instead of return? But as always: Think first.

I think I will move it inside of tileGroup.onActivation so that we are  always adding the listener (in case we change the option on the fly). It also fits with the `tileGroup.onActivation` line since it is checking for validation.